CI with Azure Pipelines

willsmythe opened the pull request:
1. Compiles client and server using Node 8 on Ubuntu (previously Node 6)
   - This matches the version of Node that VS Code builds with
   - This also enables us to use the more [efficient and predictable](321dcca1622708d4658293e87f49a56df30b12f3) "npm ci" command
2. Drops .travis.yml
3. Adds build status badge to README.md

Here is the build status (currently reflecting the statuys of my fork):

[![Build Status](https://dev.azure.com/ms/vscode-eslint/_apis/build/status/Microsoft.vscode-eslint)](https://dev.azure.com/ms/vscode-eslint/_build/latest?definitionId=18)

Once you are ready to merge this, I can work with you to update the dev.azure.com/ms/vscode-eslint pipeline to pull from Microsoft/vscode-eslint (instead of my fork).

dbaeumer commented:
@willsmythe I added the azure pipeline config file to master. What needs to be done to make this now compile in pipelines ?

willsmythe commented:
@dbaeumer - I updated the pipeline configuration on dev.azure.com/ms/vscode-eslint to pull from Microsoft/vscode-eslint now (I also added you to this project). Should be all set now.

dbaeumer commented:
@willsmythe cool. Will check it out. Can you point me to what I need to configure. Just for the future since I will now own this :-)
